blueocean-core-js/src/js/fetch.js

```diff
@@ -0,0 +1,178 @@
+     * @returns JSON body
+     */
+    rawFetchJSON(url, { onSuccess, onError, fetchOptions } = {}) {
+        const request = isoFetch(url, FetchFunctions.sameOriginFetchOption(fetchOptions))
+            .then(FetchFunctions.checkStatus)
+            .then(FetchFunctions.parseJSON);
+
+        if (onSuccess) {
+            return request.then(onSuccess).catch(FetchFunctions.onError(onError));
+        }
+        
+        return request;
+    },
+    /**
+     * Raw fetch.
+     *
+     * This method is semi-private, under normal conditions it should not be
+     * used as it does not include the JWT bearer token
+     *
+     * @param {string} url - The URL to fetch from.
+     * @param {Object} [options]
+     * @param {function} [options.onSuccess] - Optional callback success function.
+     * @param {function} [options.onError] - Optional error callback.
+     * @param {Object} [options.fetchOptions] - Optional isomorphic-fetch options.
+     * @returns fetch response
+     */
+    rawFetch(url, { onSuccess, onError, fetchOptions } = {}) {
+        const request = isoFetch(url, FetchFunctions.sameOriginFetchOption(fetchOptions))
+            .then(FetchFunctions.checkStatus);
+
+        if (onSuccess) {
+            return request.then(onSuccess).catch(FetchFunctions.onError(onError));
+        }
+      
+        return request;
+    },
+};
+
+export const Fetch = {
+    /**
+     * Fetch JSON data.
+     * <p>
+     * Utility function that can be mocked for testing.
+     *
+     * @param {string} url - The URL to fetch from.
+     * @param {Object} [options]
+     * @param {function} [options.onSuccess] - Optional callback success function.
+     * @param {function} [options.onError] - Optional error callback.
+     * @param {Object} [options.fetchOptions] - Optional isomorphic-fetch options.
+     * @returns JSON body.
+     */
+    fetchJSON(url, { onSuccess, onError, fetchOptions } = {}) {
+        return jwt.getToken()
+            .then(token => FetchFunctions.rawFetchJSON(url, {
+                onSuccess,
+                onError,
+                fetchOptions: FetchFunctions.jwtFetchOption(token, fetchOptions),
+            }));
+    },
+
+    /**
+     * Fetch data.
+     * <p>
+     * Utility function that can be mocked for testing.
+     *
+     * @param {string} url - The URL to fetch from.
+     * @param {Object} [options]
+     * @param {function} [options.onSuccess] - Optional callback success function.
+     * @param {function} [options.onError] - Optional error callback.
+     * @param {Object} [options.fetchOptions] - Optional isomorphic-fetch options.
+     * @returns fetch body.
+     */
+    fetch(url, { onSuccess, onError, fetchOptions } = {}) {
+        return jwt.getToken()
+            .then(token => FetchFunctions.rawFetch(url, {
+                onSuccess,
+                onError,
+                fetchOptions: FetchFunctions.jwtFetchOption(token, fetchOptions),
+            }));
+    },
+};
+
```

Comment:
Yes I'm aware of what they are doing. That stuff will be integrated into this during one or the merges